include/aws/s3/private/s3_client_impl.h

```diff
@@ -260,6 +260,13 @@ struct aws_s3_client {
     aws_s3_client_shutdown_complete_callback_fn *shutdown_callback;
     void *shutdown_callback_user_data;
 
+    /* Whether read backpressure (aka flow-control window) is being applied. */
+    const bool enable_read_backpressure;
+
+    /* The starting size of each meta request's flow-control window, in bytes.
+     * Ignored unless `enable_read_backpressure` is true. */
+    const size_t initial_read_window;
```

Comment:
there is no default value
If a user wants to do manual flow-control, they should pick exactly what they want this to be, since it factors into the math of how often they want to ask for more data